sympy/physics/mechanics/system.py

```diff
@@ -1,8 +1,1077 @@
-from sympy.core.backend import eye, zeros, Matrix
-from sympy.physics.vector import dynamicsymbols
-from sympy.physics.mechanics.functions import find_dynamicsymbols
+from functools import wraps
 
-__all__ = ['SymbolicSystem']
+from sympy.core.backend import Basic, ImmutableMatrix, Matrix, eye, zeros
+from sympy.core.containers import OrderedSet
+from sympy.physics.mechanics.actuator import ActuatorBase
+from sympy.physics.mechanics.body_base import BodyBase
+from sympy.physics.mechanics.functions import (
+    Lagrangian, _validate_coordinates, find_dynamicsymbols)
+from sympy.physics.mechanics.joint import Joint
+from sympy.physics.mechanics.kane import KanesMethod
+from sympy.physics.mechanics.lagrange import LagrangesMethod
+from sympy.physics.mechanics.loads import _parse_load, gravity
+from sympy.physics.mechanics.method import _Methods
+from sympy.physics.mechanics.particle import Particle
+from sympy.physics.vector import Point, ReferenceFrame, dynamicsymbols
+from sympy.utilities.iterables import iterable
+
+__all__ = ['SymbolicSystem', 'System']
+
+
+def _reset_eom_method(method):
+    """Decorator to reset the eom_method if a property is changed."""
+
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        self._eom_method = None
+        return method(self, *args, **kwargs)
+
+    return wrapper
+
+
+class System(_Methods):
+    """Class to define a system and form the equations of motion.
+
+    Explanation
+    ===========
+
+    A ``System`` instance stores the different objects associated with a model,
+    including bodies, joints, constraints, and other relevant information. With
+    all the relationships between components defined, the ``System`` can be used
+    to form the equations of motion using a backend, such as ``KanesMethod``.
+    The ``System`` has been designed to be compatible with third-party
+    libraries for greater flexibility and integration with other tools.
+
+    Attributes
+    ==========
+
+    origin : Point
+        Global origin of the system.
+    frame : ReferenceFrame
+        Inertial reference frame of the system.
+    x : Vector
+        Unit vector in the x direction of the inertial reference frame.
+    y : Vector
+        Unit vector in the y direction of the inertial reference frame.
+    z : Vector
+        Unit vector in the z direction of the inertial reference frame.
+    q : ImmutableMatrix
```

Comment:
Why Immutable only?

Reply:
Immutable is quite nice; it allows us to check the users' input. If mutable matrices are used, then users can change things without the system object knowing about it. Of course, it is possible to delay all checks to the validate system method, if that is preferred.